app/src/visionglass/cpp/DeviceDelegateVisionGlass.cpp

```diff
@@ -344,15 +357,21 @@ DeviceDelegateVisionGlass::ControllerButtonPressed(const bool aDown) {
 
 void
 DeviceDelegateVisionGlass::setHead(const float aX, const float aY, const float aZ, const float aW) {
-  // We need to flip the Z axis comming from the SDK to get the proper rotation.
-  m.headingMatrix = vrb::Matrix::Rotation({aX,aY,-aZ,-aW});
+  m.headOrientation = vrb::Quaternion(aX, aY, aZ, aW);
 }
 
 void
 DeviceDelegateVisionGlass::setControllerOrientation(const float aX, const float aY, const float aZ, const float aW) {
     m.controllerOrientation = vrb::Quaternion(aX, aY, aZ, aW);
 }
 
+void
+DeviceDelegateVisionGlass::CalibrateController() {
+  m.controllerCalibration = m.controllerOrientation.Inverse() * m.headOrientation.Inverse();
+  m.isControllerCalibrated = true;
+  m.SetupOrientationFilter();
```

Comment:
Why we need to call SetupOrientationFilter() again ? I was called when the 'm' structure is instantiated, wasn't it ?

Reply:
We need to reset it. Otherwise the filter will use the previous uncalibrated data to perform the filtering. The result is horrible because it's trying to compensate the previous deviation.